### PR TITLE
Generate linting errors via JSHint

### LINKED ISSUE #4712: [Bug] JS errors on release seem totally random
@Nikhil-Nandagopal mentioned that he keeps seeing this even though there is no semicolon in his code

@hetunandu  mentioned that 
This is happening because the final code that gets evaluated is
```
function() {
  const result = appsmith.mode.charAt(; // <- Syntax error
  return { result, triggers: self.triggers }
}()
```
This is confusing because the user never wrote the whole function but is getting a syntax error.

### LINKED ISSUE #4977: [Feature] Generate linting errors and warnings during evaluation
## Summary
Many errors in code could be avoided if there were lining errors that are shown to the user. Showing these errors and warnings ensures users follow best practices while writing code

## Additional Context
While ESlint cannot be run on the browser we can use JShint to find these linting errors

### LINKED ISSUE #5109: [Feature] Create a `bindingSuccess` event
## Summary
Track when binding by the user was successful and valid

Track:
- widgetType
- propertyName
- unEval value

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Generating lining errors via JSHint
- Not showing these errors right now as the UI is still missing
- Collated all the different errors that can be generated via evaluations under single `__evaluataions__.errors` list
- Set `__evaluataions__.values` for actions and widget properties

Fixes #5109 
Fixes #4977
Fixes #4712

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/create-js-linting-errors 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 53.38 **(0.02)** | 35.4 **(0.07)** | 32.01 **(0.06)** | 53.91 **(0.03)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/EvaluatedValuePopup.tsx | 63.95 **(-0.25)** | 61.29 **(4.37)** | 41.38 **(-1.48)** | 66.25 **(-0.42)**
 :red_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 74.02 **(-1.33)** | 50.43 **(-2.46)** | 58.06 **(0)** | 74.87 **(-1.13)**
 :red_circle: | app/client/src/sagas/EvaluationsSaga.ts | 17.5 **(-1.69)** | 1.3 **(-0.26)** | 7.69 **(-1.4)** | 20.35 **(-1.8)**
 :green_circle: | app/client/src/sagas/OnboardingSagas.ts | 18.36 **(0.28)** | 3.7 **(0.13)** | 12.24 **(-0.26)** | 18.71 **(0.32)**
 :green_circle: | app/client/src/selectors/propertyPaneSelectors.tsx | 96.15 **(18.73)** | 94.12 **(17.93)** | 100 **(0)** | 96 **(19.33)**
 :red_circle: | app/client/src/utils/DynamicBindingUtils.ts | 88.46 **(0.22)** | 76.06 **(0.65)** | 71.43 **(-7.14)** | 86.73 **(-0.08)**
 :red_circle: | app/client/src/widgets/BaseWidget.tsx | 78.81 **(-1.86)** | 41.86 **(0)** | 74.19 **(-5.81)** | 77.88 **(-1.77)**
 :green_circle: | app/client/src/widgets/ListWidget/ListPropertyPaneConfig.ts | 50 **(4.55)** | 0 **(0)** | 0 **(0)** | 50 **(4.55)**
 :green_circle: | app/client/src/widgets/Tabs/TabsMigrator.tsx | 75 **(1.09)** | 0 **(0)** | 40 **(0)** | 73.91 **(1.18)**
 :green_circle: | app/client/src/workers/DataTreeEvaluator.ts | 88.61 **(-0.1)** | 79.27 **(0.89)** | 87.32 **(0)** | 88.79 **(0)**
 :green_circle: | app/client/src/workers/evaluate.ts | 100 **(0)** | 70 **(3.33)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/workers/evaluationUtils.ts | 89.63 **(0.74)** | 75.89 **(-0.53)** | 90.91 **(0.21)** | 89.37 **(0.7)**</details>

### CODE REVIEW

**Comment by SatishGandham** [comment]
Looks good wrt to my performance related changes.

File: app/client/src/components/editorComponents/CodeEditor/EvaluatedValuePopup.tsx [hunk 1]
```diff
@@ -267,30 +269,43 @@ export const CurrentValueViewer = memo(
...
+      {hasError && error && (
         <ErrorText>
-          <span className="t--evaluatedPopup-error">
-            {props.jsError && props.jsError.length
-              ? props.jsError
-              : props.useValidationMessage && props.error
-              ? props.error
-              : `This value does not evaluate to type "${props.expected}". Transform it using JS inside '{{ }}'`}
-          </span>
+          {error.errorType.toLowerCase()} error:&nbsp;
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/CodeEditor/EvaluatedValuePopup.tsx:297]: remove this

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

**Review by ApekshaBhosale — DISMISSED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Review by ApekshaBhosale — DISMISSED** [review]

**Pull request merged**